### PR TITLE
Add diagnostics to Go API reviews

### LINKED ISSUE #3059: Show exported fields of embedded unexported structs
Given this code:
```go
type unexported struct {
    X int
}

type Exported struct {
    unexported
}
```
APIView will show `Exported` as having no fields despite `Exported.X` being accessible. I think we shouldn't write code like this because it makes a confusing API--`Exported.X` can't be set in a literal--but it would help for APIView to make this pattern visible. The solution requires some design thought. How should this pattern look in an API review?

### LINKED ISSUE #3105: Fix navigation links to types having the same name
Navigation anchors use unqualified type names, making it impossible to link to two types in different packages which have the same name, for example `azservicebus.Client` and `azservicebus/atom.Client`. Fixing this requires qualifying links with at least package names, if not module names. Related: #3049

### LINKED ISSUE #3152: Add diagnostics to Go reviews
Two interesting cases to start with:
1. anonymously embedding an unexported struct with exported fields is an error
     - #3059
2. types exported by alias from an `internal` package should be flagged with an info or warning diagnostic pointing this out

### PR DESCRIPTION
**Pull request by chlowell**
My main goal here is to add diagnostics that surface otherwise invisible information about public APIs (closes #3152):
- when a type is an alias for another
- when a struct anonymously embeds an unexported struct (closes #3059)
- when an interface can't be implemented by users

This is a big change because these diagnostics require inspecting unexported nodes. The tool was designed to ignore these because we never want unexported types to appear in reviews. Keeping them out now requires more care. I also did some refactoring to make the code easier to test and, I hope, maintain.

Also fixed some bugs along the way:
- duplicate symbol IDs (fixes #3105)
- several instances of unhandled syntax
- aliases whose source can't be found don't appear in reviews
- panic on unnamed parameters

### CODE REVIEW

File: src/go/cmd/api_view.go [hunk 1]
```diff
@@ -13,12 +13,32 @@ import (
...
 // CreateAPIView generates the output file that the API view tool uses.
 func CreateAPIView(pkgDir, outputDir string) error {
-	m, err := NewModule(pkgDir)
+	review, err := createReview(pkgDir)
 	if err != nil {
 		panic(err)
 	}
+	if outputDir == "." {
+		outputDir = ""
+	} else if !strings.HasSuffix(outputDir, "/") {
```
Comment (jhendrixMSFT) [inline src/go/cmd/api_view.go:22]: Does `outputDir` have its path separator chars normalized elsewhere?

File: src/go/cmd/api_view.go [hunk 2]
```diff
@@ -13,12 +13,32 @@ import (
...
+		outputDir = fmt.Sprintf("%s/", outputDir)
+	}
+	filename := fmt.Sprintf("%s%s.json", outputDir, review.Name)
```
Comment (jhendrixMSFT) [inline src/go/cmd/api_view.go:25]: Another option, which might avoid the need to normalize path separators, is to use [`filepath.Join`](https://pkg.go.dev/path/filepath#Join).

**Comment by jhendrixMSFT** [comment]
Is there a link to a review with all of these changes?

**Comment by chlowell** [comment]
Examples:
- [azblob](https://apiviewuat.azurewebsites.net/Assemblies/Review/41795fedb17e49c388f2a01fc112e241)
- [azcore](https://apiviewuat.azurewebsites.net/Assemblies/Review/a8f019cb3af34a269c738ea2684386cd)
- [azidentity](https://apiviewuat.azurewebsites.net/Assemblies/Review/7d6397487cd147b7a5887f5b8973078a)
- [azservicebus](https://apiviewuat.azurewebsites.net/Assemblies/Review/89499199bbe54295973d78fe261d16d8)

**Comment by jhendrixMSFT** [comment]
Is this an issue with the generated doc or the UI?

image

**Comment by jhendrixMSFT** [comment]
Can we indent package content, so the outline is clear?  Feel free to add this in a separate PR.

**Comment by jhendrixMSFT** [comment]
Also, at some point, we should move the sources to an `apiview` subdirectory in case we add other tools for Go.

**Comment by jhendrixMSFT** [comment]
Another idea for future enhancement, add a diagnostic for deprecated content.

**Comment by chlowell** [comment]
> Is this an issue with the generated doc or the UI?
> 

The generated doc. In these cases the tool copies expressions from source files.

> Can we indent package content, so the outline is clear? Feel free to add this in a separate PR.
>

We can, at the cost of more horizontal scrolling, Maybe we should also format funcs like structs, one parameter per line?

**Comment by jhendrixMSFT** [comment]
While I'm not a fan of the one-parameter-per-line when writing code, it might make sense for APIView.  I don't have a strong opinion for this one, so I'll defer to @JeffreyRichter.

**Comment by JeffreyRichter** [comment]
I agree with Joel. I've never seen any code source code with 1 param per line; it i snot idiomatic.

**Comment by chlowell** [comment]
To be clear, I meant we could format parameters that way in API reviews, whose formatting needn't match the source's. I agree it would look unconventional.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**